app/models/metric/ci_mixin/capture.rb

```diff
@@ -78,15 +89,17 @@ def perf_capture_queue(interval_name, options = {})
         if msg.nil?
           qi[:priority] = priority
           qi.delete(:state)
-          qi[:miq_callback] = cb if cb
+          if cb && item_interval == "realtime"
+            qi[:miq_callback] = cb
+          end
           qi
         elsif msg.state == "ready" && (task_id || MiqQueue.higher_priority?(priority, msg.priority))
           qi[:priority] = priority
           # rerun the job (either with new task or higher priority)
           qi.delete(:state)
           if task_id
             existing_tasks = (((msg.miq_callback || {})[:args] || []).first) || []
-            qi[:miq_callback] = cb.merge(:args => [existing_tasks + [task_id]])
+            qi[:miq_callback] = cb.merge(:args => [existing_tasks + [task_id]]) if item_interval == "realtime"
```

Comment:
If we can get rid of the check on upgrading priority, then it will be easier to go over to `put_unless_exists`.

Real-time will potentially still need this